Doc for Query Builder (#110)
biyard/rust-sdk: docs/common/models/query-builder.md

# Query Builder

> The Query Builder is a powerful tool for constructing database queries in a type-safe and fluent manner. It's automatically generated for models marked with the #[api_model] attribute when the "server" feature is enabled. The builder provides a chainable interface for constructing complex queries with conditions, ordering, and other database operations.

Reviewer: ```suggestion
The Query Builder is a powerful tool for constructing database queries in a type-safe and fluent manner. It's automatically generated for models marked with the `#[api_model]` attribute when the `server` feature is enabled. The builder provides a chainable interface for constructing complex queries with conditions, ordering, and other database operations.
```

## Getting Started

To use the Query Builder for a model, first ensure your model is properly annotated:

```rust

> #[api_model(queryable)]

Reviewer: ```suggestion
#[api_model(base = "/v1/endpoint", table = my_models)]
```

struct MyModel {
    id: u64,
    name: String,
    is_active: bool,
    created_at: DateTime<Utc>,
}
```

Then you can access the query builder through the model's query_builder() method:

```rust
let query = MyModel::query_builder();
```

## Building Queries

### Basic Query Construction

The Query Builder supports a fluent interface for constructing queries:

```rust
let query = MyModel::query_builder()

>     .name_equals("John Doe")

Reviewer: ```suggestion
    .name_equals("John Doe".to_string())
```

# Query Builder

> The Query Builder is a powerful tool for constructing database queries in a type-safe and fluent manner. It's automatically generated for models marked with the '#[api_model]' attribute when the 'server' feature is enabled. The builder provides a chainable interface for constructing complex queries with conditions, ordering, and other database operations.

Reviewer: ```suggestion
The Query Builder is a powerful tool for constructing database queries in a type-safe and fluent manner. It's automatically generated for models marked with the `#[api_model]` attribute when the `server` feature is enabled. The builder provides a chainable interface for constructing complex queries with conditions, ordering, and other database operations.
```

This is inline code block of markdown syntax
